### ISSUE TITLE
Operator CI failing due to failures while pulling base image

### ISSUE THREAD

**Issue by vmuzikar** [keycloak/keycloak#28036]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

operator

### Describe the bug

Often fails with:
```
failed to solve: registry.access.redhat.com/ubi9-micro: pulling from host registry.access.redhat.com failed with status code [manifests latest]: 503 Service Unavailable
```
https://github.com/keycloak/keycloak/actions/runs/8340088871/job/22823672779?pr=28031

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

CI passes

### Actual behavior

CI intermittently fails

### How to Reproduce?

Run CI

**Comment by vmuzikar** [comment]
Did 20 image builds/test runs in my own fork and 10 runs in my private fork (which uses smaller GHA runners). All passed. I'd assume it was just some temporary network issue.

I'll keep this open and further monitor the CI if it happens again.

**Comment by vmuzikar** [comment]
In parallel, @Pepo48 will try and find the stakeholders for the registry to discuss what could be the potential cause and if it's a known issue.

**Comment by Pepo48** [comment]
I went checked the outage history and existing support cases - none of them mentions our specific problem (yet multiple similar issues from the past suggest that this may be something temporary). I also spoke with folks outside of our team, seeking for further info. They confirmed that it seems like something temporary.

I ended up with filling a support case that should be forwarded to the RH registry maintainers.
